tests/host_tools/change_config_space.c

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// This is a binary, used by the Firecracker `test_config_space.py`
+// integration test. It writes into the microVM configured network
+// device config space a new MAC address.
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+int main(int argc, char *argv[]) {
+
+    int fd;
+    uint8_t *map_base, *virt_addr;
+
+    uint64_t mapped_size, page_size, offset_in_page, target;
+    uint64_t width = 6;
+
+    uint64_t config_offset = 0x00000000;
+    uint64_t mmio_start_address = 0x00000000;
+
+#ifdef __x86_64__
+    mmio_start_address = 0xD0000000;
+    /* The offset in MMIO space where the config space
+     * for the net device starts.
+     * Net device config space offset. From 0xD0000000
+     * to 0xD0001000 is the MMIO space for the root device
+     * (block device). */
+    config_offset = 0x00001100;
+#endif
+
+#ifdef __aarch64__
+    mmio_start_address = 0x40001000;
+    /* The offset in MMIO space where the config space
```

Comment:
The config offset is the same on both platform (0x100) as specified by the virtio specification. What is changed on different platforms is the location of the devices on the bus. This should be reflected in the code as well.

I think @sandreim had a nice suggestion of finding the start address of a specific device using information from `/proc`. This is more of a nit, but we might get unexpected failures if we switch the order in which we are adding devices on the bus in Firecracker.

Reply:
```bash
cat /proc/cmdline
console=ttyS0 reboot=k panic=1 pci=off root=/dev/vda rw virtio_mmio.device=4K@0xd0000000:5 virtio_mmio.device=4K@0xd0001000:6 virtio_mmio.device=4K@0xd0002000:7
```

```bash
cat /sys/devices/virtio-mmio-cmdline/virtio-mmio.2/virtio2/uevent
DRIVER=virtio_net
MODALIAS=virtio:d00000001v00000000
```

Reply:
If I remember correctly on arm it was harder to take this infos from the guest.

Reply:
That is correct @lauralt . That information lives in the FDT. @iulianbarbu you can try [fdt python module](https://pypi.org/project/fdt/) or something similar.

Reply:
@sandreim, @andreeaflorescu , PTAL! I've addressed your feedback.